content-script.js

```diff
@@ -48,49 +55,62 @@ var code = `
 			+ 'width: 45px;'
 			+ 'margin: 5px;';
 		input.value = lastSpeed * 100;
-		input.oninput = function(e){ /* What happens when we change the number in our input box element */
-			validateAndChangeSpeed();  /* We call our function */
+		input.onkeypress = function(e){ /* What happens when we change the number in our input box element */
+			if (e.code === "Enter") {
+				validateAndChangeSpeed();
+			}
+		};
+		input.onblur = function() {
+			// "onfocusout" event not working
+			validateAndChangeSpeed();
 		};
 		
 		function validateAndChangeSpeed(value){ 
-			var val = parseFloat( value || (input.value / 100)); /* val must be in format 0.0625 - 16.0 https://stackoverflow.com/a/32320020 */
-			if(!isNaN(val)){ /* check if val is a number */
-				changeSpeed(val);
+			var val = parseFloat( value || (input.value / 100));
+			if (!isNaN(val) && (val !== lastSpeed)) {
+				/* only change if input is valid and it changed */
+				setStoredSpeed(val);
+				videosChangeSpeed(val);
 			}
 		}
 		
-		function changeSpeed(val) {
+		function videosChangeSpeed(val) {
 			for(var i = 0; i < VideoElementsMade.length; i++){ /* change speed for all elements found (i havent seen this be more than 1 but you never know) */
-				VideoElementsMade[i].playbackRate = val; /* set the playback rate here */
-				if(val != lastSpeed){ /* update the lastSpeed if the speed actually changed */
-					lastSpeed = val;
-					setStoredSpeed(val);
+				/* val is clamped to range 0.0625 - 16.0 https://stackoverflow.com/a/32320020 */
+				if (VideoElementsMade[i].playbackRate !== val) {
+					debugLog("changing playback rate from " + VideoElementsMade[i].playbackRate + " to " + val)
 				}
+				VideoElementsMade[i].playbackRate = val; /* set the playback rate here */
+				VideoElementsMade[i].defaultPlaybackRate = val;
 			}
 		}
 		
-		function timeout() { /* This function is called by itself over and over */
-			if(document.getElementById('speed-extension-input') == null) /* check if our input element doesnt exist */
-			{
-				try {
-					document.getElementsByClassName('now-playing-bar__right')[0].appendChild (input); /* make our input exist on page */
-				}catch{
-					setTimeout(timeout, 100);/*now-playing-bar__right doesnt exist yet so lets try again in 100ms*/
-					return;
-				}
+		function addSpeedInput() { /* adds speed input next to volume bar */
+			debugLog("Adding speed input");
+			try {
+				document.getElementsByClassName('volume-bar')[0].appendChild(input); /* make our input exist on page */
+				debugLog("Added speed input");
+			}catch{
+				setTimeout(addSpeedInput, 100);/*volume-bar doesnt exist yet so lets try again in 100ms*/

```

Comment:
```suggestion
				setTimeout(addSpeedInput, 1000);/*volume-bar doesnt exist yet so lets try again in 1 second*/
```
retrying every 100 ms is quite excessive; retry only every second